src/app/fdctl/run/tiles/fd_sign.c

```diff
@@ -121,6 +127,29 @@ after_frag( void *             _ctx,
       fd_ed25519_sign( ctx->out[ in_idx ].data, ctx->_data, 130UL, ctx->public_key, ctx->private_key, ctx->sha512 );
       break;
     }
+    case FD_KEYGUARD_ROLE_GOSSIP: {
+      if( FD_UNLIKELY( !fd_keyguard_payload_authorize( ctx->_data, *opt_sz, FD_KEYGUARD_ROLE_GOSSIP ) ) ) {
+        FD_LOG_EMERG(( "fd_keyguard_payload_authorize failed %lu %u %u %u %u", *opt_sz, ctx->_data[0], ctx->_data[1], ctx->_data[2], ctx->_data[3] ));
```

Comment:
Can you remove the dereferences? They might not be valid here and its' another chance to corrupt the tile